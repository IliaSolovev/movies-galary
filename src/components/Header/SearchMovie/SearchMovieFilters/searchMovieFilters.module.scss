.searchMovieFilters{
  width: 250px;
  margin-bottom: 5px;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}
.searchBy{
  font-family: Roboto,sans-serif;
  text-transform: uppercase;
  color: #fff;
}
.buttons{
  width: 57%;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}