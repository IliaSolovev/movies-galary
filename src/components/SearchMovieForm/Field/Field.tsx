import React from "react";
import s from './field.module.scss'
interface Props {
    value: string,
    onChange: (e: React.FormEvent ) => void
}

const Field = (props: Props) => {
    // const {value,onChange} = props;
    return (
        <>
            <input type="text" className={s.input} />
        </>
    )

};

export default Field;