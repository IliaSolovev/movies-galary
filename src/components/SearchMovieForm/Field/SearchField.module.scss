@import "../../../styles/variables";

.searchField {
  position: relative;
  width: 100%;
  height: 40px;
  margin-bottom: 15px;
  padding: 10px;
  background: rgba(0, 0, 0, 0.84);
  border-radius: 5px 5px 0  0;
  border: none;
  border-bottom: 2px solid $redColor;
  font-family: $mainFont;
  font-weight: lighter;
  font-size: 1em;
  color: #fff;
  box-sizing: border-box;
}
