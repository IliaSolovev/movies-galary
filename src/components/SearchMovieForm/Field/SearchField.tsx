import React, { ChangeEvent } from 'react';

import style from './searchField.module.scss';

interface Props {
    value: string,
    onChange: (value: string) => void
}

export const SearchField:React.FC<Props> = ({ value, onChange }) => (
  <>
    <input
      type="text"
      className={style.searchField}
      value={value}
      onChange={({ currentTarget }:ChangeEvent<HTMLInputElement>) => onChange(currentTarget.value)}
    />
  </>
);
