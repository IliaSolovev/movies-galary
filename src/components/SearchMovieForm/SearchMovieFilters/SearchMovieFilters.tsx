import React from "react";
import SearchMovieFiltersButton from "./SearchMovieFiltersButton/SearchMovieFiltersButton";
import s from "./searchMovieFilters.module.scss"
interface Props {

}

const SearchMovieFilters = (props: Props) => {
    return (
        <div className={s.searchMovieFilters}>
            <div className={s.searchBy}>search by</div>
            <div className={s.buttons}>
                <SearchMovieFiltersButton text="title" isActive={true}/>
                <SearchMovieFiltersButton text="genre" isActive={false}/>
            </div>
        </div>
    )
};

export default SearchMovieFilters