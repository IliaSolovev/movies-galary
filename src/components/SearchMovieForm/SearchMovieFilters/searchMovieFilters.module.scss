@import '../../../styles/variables';
.searchMovieFilters{
  width: 250px;
  margin-bottom: 5px;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  &__searchBy{
    font-family: $mainFont;
    text-transform: uppercase;
    color: #fff;
  }
  &__actions{
    width: 57%;
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
  }
}
