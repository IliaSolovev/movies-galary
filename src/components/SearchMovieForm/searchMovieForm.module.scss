@import "./../../styles/main";
@import "./../../styles/variables";
.searchMovie{
  position: relative;
  @extend .headerContainer;
}
.searchMovie__title{
  margin-bottom: 1.4vw;
  font-family: $mainFont;
  font-weight: bold;
  color: #fff;
  text-transform: uppercase;
}

.searchMovie__buttons{
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  @media screen and (max-width: 500px) {
    justify-content: space-around;

  }
}