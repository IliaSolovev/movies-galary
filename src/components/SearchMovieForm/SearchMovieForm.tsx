import React from 'react';

import { SearchField } from './Field/SearchField';
import { SearchMovieFilters } from './SearchMovieFilters/SearchMovieFilters';
import { SearchType } from '../../redux/moviesSlice';
import { Button } from '..';

import style from './searchMovieForm.module.scss';

interface Props {
    fieldValue: string,
    onFieldChange: (value: string) => void,
    searchType: string,
    onSelectType: (value: SearchType) => void,
    onSearch: () => void
}

export const SearchMovieForm: React.FC<Props> = ({
  fieldValue, onFieldChange, searchType, onSelectType, onSearch,
}) => (
  <div className={style.searchMovie}>
    <div className={style.searchMovie__title}>find your movie</div>
    <div>
      <SearchField value={fieldValue} onChange={onFieldChange} />
    </div>
    <div className={style.searchMovie__buttons}>
      <SearchMovieFilters searchType={searchType} onSelectType={onSelectType} />
      <Button onClick={onSearch} type="search">search</Button>
    </div>
  </div>
);
