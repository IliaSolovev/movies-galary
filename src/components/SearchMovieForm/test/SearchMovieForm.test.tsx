import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SearchField } from '../Field/SearchField';
import { SearchMovieFilters } from '../SearchMovieFilters/SearchMovieFilters';
import { SearchMovieForm } from '../SearchMovieForm';
import { Button } from '../..';

it('should render search field', () => {
  const component = shallow(<SearchField value="asd" onChange={() => {}} />);
  expect(toJson(component)).toMatchSnapshot();
});

it('should render search movie filter button', () => {
  const component = shallow(<Button onClick={() => {}} active type="search">search</Button>);
  expect(toJson(component)).toMatchSnapshot();
});
it('should render search movie filter ', () => {
  const component = mount(<SearchMovieFilters searchType="genres" onSelectType={() => {}} />);
  expect(toJson(component)).toMatchSnapshot();
});
it('should render search movie form ', () => {
  const component = mount(<SearchMovieForm
    searchType="title"
    fieldValue="asd"
    onFieldChange={() => {}}
    onSelectType={() => {}}
    onSearch={() => {}}
  />);
  expect(toJson(component)).toMatchSnapshot();
});
