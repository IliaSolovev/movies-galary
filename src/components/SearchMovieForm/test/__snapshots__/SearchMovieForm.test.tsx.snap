// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render search field 1`] = `
<Fragment>
  <input
    className="searchField"
    onChange={[Function]}
    type="text"
    value="asd"
  />
</Fragment>
`;

exports[`should render search movie filter  1`] = `
<Component
  onSelectType={[Function]}
  searchType="genre"
>
  <div
    className="searchMovieFilters"
  >
    <div
      className="searchMovieFilters__searchBy"
    >
      search by
    </div>
    <div
      className="searchMovieFilters__buttons"
    >
      <Component
        active={false}
        key="0"
        onClick={[Function]}
        type="searchFilter"
      >
        <button
          className="button button__searchFilter "
          onClick={[Function]}
        >
          title
        </button>
      </Component>
      <Component
        active={true}
        key="1"
        onClick={[Function]}
        type="searchFilter"
      >
        <button
          className="button button__searchFilter button__searchFilter_active"
          onClick={[Function]}
        >
          genre
        </button>
      </Component>
    </div>
  </div>
</Component>
`;

exports[`should render search movie filter button 1`] = `
<button
  className="undefined  undefined"
  onClick={[Function]}
>
  asd
</button>
`;

exports[`should render search movie form  1`] = `
<Component
  fieldValue="asd"
  onFieldChange={[Function]}
  onSelectType={[Function]}
  search={[Function]}
  searchType="title"
>
  <div
    className="searchMovie"
  >
    <div
      className="searchMovie__title"
    >
      find your movie
    </div>
    <div>
      <Component
        onChange={[Function]}
        value="asd"
      >
        <input
          className="searchField"
          onChange={[Function]}
          type="text"
          value="asd"
        />
      </Component>
    </div>
    <div
      className="searchMovie__buttons"
    >
      <Component
        onSelectType={[Function]}
        searchType="title"
      >
        <div
          className="searchMovieFilters"
        >
          <div
            className="searchMovieFilters__searchBy"
          >
            search by
          </div>
          <div
            className="searchMovieFilters__buttons"
          >
            <Component
              active={true}
              key="0"
              onClick={[Function]}
              type="searchFilter"
            >
              <button
                className="button button__searchFilter button__searchFilter_active"
                onClick={[Function]}
              >
                title
              </button>
            </Component>
            <Component
              active={false}
              key="1"
              onClick={[Function]}
              type="searchFilter"
            >
              <button
                className="button button__searchFilter "
                onClick={[Function]}
              >
                genre
              </button>
            </Component>
          </div>
        </div>
      </Component>
      <Component
        onClick={[Function]}
        type="search"
      >
        <button
          className="button button__search "
          onClick={[Function]}
        >
          search
        </button>
      </Component>
    </div>
  </div>
</Component>
`;
